src/mqt/bench/benchmark_generation.py

```diff
@@ -414,7 +413,7 @@ def get_benchmark(
     circuit_size: int | None = None,
     benchmark_instance_name: str | None = None,
     compiler_settings: CompilerSettings | None = None,
-    gateset: str | Gateset = "ibm_falcon",
+    gateset: str | Target = "ibm_falcon",
     device_name: str = "ibm_washington",
```

Comment:
Just to be sure. This should also become one parameter here.
And I am fairly sure, that we do not want to allow strings here anymore, but only real `Target`s.
We can provide a separate convenience method to instantiate targets by name, but I would argue it helps the interface to reduce the potential typing differences.